nanoFramework.CoreLibrary/System/ArgumentNullException.cs

```diff
@@ -11,23 +16,54 @@ namespace System
         /// </summary>
-        public ArgumentNullException()
-        { }
+        /// <param name="paramName">The name of the parameter that caused the exception.</param>
+        public ArgumentNullException(string? paramName) : base(ArgumentNull_Generic, paramName)
+        {
+        }
 
         /// <summary>
-        /// Initializes a new instance of the ArgumentNullException class with the name of the parameter that causes this exception.
+        /// Initializes a new instance of the <see cref="ArgumentNullException"/> class with a specified error message and the exception that is the cause of this exception.
         /// </summary>
-        /// <param name="argument">The name of the parameter that caused the exception.</param>
-        public ArgumentNullException(string argument) : base(null, argument) { }
+        /// <param name="message">The error message that explains the reason for this exception.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception, or a null reference (Nothing in Visual Basic) if no inner exception is specified.</param>
+        public ArgumentNullException(string? message, Exception? innerException) : base(message ?? ArgumentNull_Generic, innerException)
+        {
+        }
 
         /// <summary>
-        /// Initializes an instance of the ArgumentNullException class with a specified error message and the name of the parameter that causes this exception.
+        /// Initializes an instance of the <see cref="ArgumentNullException"/> class with a specified error message and the name of the parameter that causes this exception.
         /// </summary>
-        /// <param name="paramName">The name of the parameter that caused the exception. </param>
-        /// <param name="message">A message that describes the error. </param>
-        public ArgumentNullException(String paramName, String message) : base(message, paramName) { }
+        /// <param name="paramName">The name of the parameter that caused the exception.</param>
+        /// <param name="message">The error message that explains the reason for this exception.</param>
+        public ArgumentNullException(string? paramName, string? message) : base(message ?? ArgumentNull_Generic, paramName) { }
+
+        /// <summary>Throws an <see cref="ArgumentNullException"/> if <paramref name="argument"/> is null.</summary>
+        /// <param name="argument">The reference type argument to validate as non-null.</param>
+        /// <param name="paramName">The name of the parameter with which <paramref name="argument"/> corresponds.</param>
+        public static void ThrowIfNull([NotNull] object? argument, [CallerArgumentExpression(nameof(argument))] string? paramName = null)
+        {
+            if (argument is null)
+            {
+                Throw(paramName);
+            }
+        }
+
+        [DoesNotReturn]
+        internal static void Throw(string? paramName) =>
```

Comment:
Same as the other. If the extra method is a concern I can roll it into the primary call.